### PR TITLE
Feature/INTEG-86: Add listAccountSummaries Endpoint and Handle API errors

### PR DESCRIPTION
**Pull request by pkeavenycontentful**
## Purpose
We want to provide actionable feedback to space admins if their service account is missing access to any Google Analytics properties.

## Approach
I adjusted some of the error handling on the lambda side so we could destructure the data/errors out of the `fetch()` call. Also piggy backed on the current errors we have on the lambda side instead of custom ones, with the exception of the case where the service account user is not added to any analytics dashboards yet, in which case we have a custom error text on the frontend for this.

### CODE REVIEW

File: apps/google-analytics-4/lambda/src/app.ts [hunk 1]
```diff
@@ -70,18 +70,23 @@ app.get('/api/credentials', (_req, res) => {
...
+      throw new Error('missing service account key value');
+    }
 
-  const googleApi = GoogleApi.fromServiceAccountKeyFile(serviceAccountKeyFile);
-  const result = await googleApi.listAccountSummaries();
-  res.status(200).json(result);
+    const googleApi = GoogleApi.fromServiceAccountKeyFile(serviceAccountKeyFile);
+    const result = await googleApi.listAccountSummaries();
+    res.status(200).json(result);
+  } catch (err) {
```
[the remark this answers is not in the document]
Comment (pkeavenycontentful) [reply apps/google-analytics-4/lambda/src/app.ts:86]: Feel free to investigate and let me know what you come up with 👍 . FYI here's the docs I followed to get this working: https://expressjs.com/en/guide/error-handling.html

File: apps/google-analytics-4/frontend/src/components/InstalledServiceAccountKey.tsx [hunk 2]
```diff
@@ -53,9 +56,13 @@ const InstalledServiceAccountKey = ({
   useEffect(() => {
     (async () => {
       try {
-        const response = await api.getCredentials();
-        setCredentialsData(response);
-        setError(null);
+        const credentials = await api.getCredentials();
```
[the remark this answers is not in the document]
Comment (pkeavenycontentful) [reply apps/google-analytics-4/frontend/src/components/InstalledServiceAccountKey.tsx]: Removed. I also removed the state variables and just set the Note to show Active as long as there isn't an error and at least one account summary is present.

File: apps/google-analytics-4/frontend/src/components/InstalledServiceAccountKey.tsx [hunk 3]
```diff
@@ -53,9 +56,13 @@ const InstalledServiceAccountKey = ({
...
+        const accountSummaries = await api.listAccountSummaries();
+        setCredentialsData(credentials);
+        if (accountSummaries.length === 0 ) {
+          throw new ApiError("You need to make sure your service account is added to the property you want to connect this space to. Right now, your service account isn't connected to any Analytics properties.")
```
[the remark this answers is not in the document]
Comment (pkeavenycontentful) [reply apps/google-analytics-4/frontend/src/components/InstalledServiceAccountKey.tsx:60]: Is there a specific guide you used to do this? Looking through the docs and want to make sure I get the right one. Here's the closest I found: https://support.google.com/analytics/answer/1009694?hl=en

[the remark this answers is not in the document]
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/InstalledServiceAccountKey.tsx:60]: This is probably the link that's closest: https://support.google.com/analytics/answer/9305788?hl=en&ref_topic=12154331,12153943,2986333,&visit_id=638120000716667257-2222825247&rd=1#zippy=%2Cin-this-article

Feel free to stick the URL in there for now, but i think we should really "tackle"  this in a subsequent task / PR.

**Review by jsdalton — APPROVED** [review]
💯

**Pull request merged**